subprojects/persistent-cache/src/test/groovy/org/gradle/cache/internal/ProducerGuardTest.groovy

```diff
@@ -45,7 +47,36 @@ class ProducerGuardTest extends ConcurrentSpec {
         calls.get() == 100
 
         where:
-        guard << [ProducerGuard.serial(), ProducerGuard.striped(), ProducerGuard.adaptive()]
+        guard << [ProducerGuard.serial(), ProducerGuard.striped(), ProducerGuard.adaptive(), ProducerGuard.timingOut(10, TimeUnit.SECONDS)]
```

Comment:
Can't we use some smaller value for the timeout in this test? 10 seconds seems to be pretty long.

Reply:
It's long but we never reach that point. It's if the test takes 10 seconds that we have a problem. The test which actually checks if a timeout is reached uses 10 ms.